plugin/evm/vm.go

```diff
@@ -703,12 +703,41 @@ func (vm *VM) initializeChain(lastAcceptedHash common.Hash) error {
 	// Set the gas parameters for the tx pool to the minimum gas price for the
 	// latest upgrade.
 	vm.txPool.SetGasTip(big.NewInt(0))
-	vm.txPool.SetMinFee(big.NewInt(params.ApricotPhase4MinBaseFee))
+	vm.setMinFeeAtEUpgrade()
 
 	vm.eth.Start()
 	return vm.initChainState(vm.blockChain.LastAcceptedBlock())
 }
 
+// TODO: remove this after EUpgrade is activated
+func (vm *VM) setMinFeeAtEUpgrade() {
+	now := vm.clock.Time()
+	if vm.chainConfig.EUpgradeTime == nil {
+		// If EUpgrade is not set, set the min fee according to the latest upgrade
+		vm.txPool.SetMinFee(big.NewInt(params.ApricotPhase4MinBaseFee))
+		return
+	} else if vm.chainConfig.IsEUpgrade(uint64(now.Unix())) {
+		// If EUpgrade is activated, set the min fee to the EUpgrade min fee
+		vm.txPool.SetMinFee(big.NewInt(params.EUpgradeMinBaseFee))
+		return
+	}
+
+	vm.txPool.SetMinFee(big.NewInt(params.ApricotPhase4MinBaseFee))
+	vm.shutdownWg.Add(1)
+	go func() {
```

Comment:
I left it here since it is temporary anyway, can be factored out to its own file if others prefer.